src/components/CommitGraph/index.tsx

```diff
@@ -120,6 +122,7 @@ export default function CommitGraph({
                 dateFormatFn={dateFormatFn}
                 currentBranch={currentBranch}
                 fullSha={fullSha}
+                onClick={() => onClick && onClick(commits[index])}
```

Comment:
avoid create an onClick function when not necessary.
```suggestion
                onClick={onClick ? () => onClick(commits[index]) : undefined}
```